pkg/serializer/json_payload_builder_benchmark_test.go

```diff
@@ -0,0 +1,89 @@
+//+build zlib
+
+package serializer
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/DataDog/datadog-agent/pkg/serializer/stream"
+)
+
+func benchmarkJSONPayloadBuilderThroughput(points int, items int, tags int, runs int) { //nolint:unuse
+	series := metrics.Series{}
+	for i := 0; i < items; i++ {
+		series = append(series, &metrics.Serie{
+			Points: func() []metrics.Point {
+				ps := make([]metrics.Point, points)
+				for p := 0; p < points; p++ {
+					ps[p] = metrics.Point{Ts: float64(p * i), Value: float64(p + i)}
+				}
+				return ps
+			}(),
+			MType:    metrics.APIGaugeType,
+			Name:     "test.metrics",
+			Interval: 15,
+			Host:     "localHost",
+			Tags: func() []string {
+				ts := make([]string, tags)
+				for t := 0; t < tags; t++ {
+					ts[t] = fmt.Sprintf("tag%d:foobar", t)
+				}
+				return ts
+			}(),
+		})
+	}
+	json, _ := series.MarshalJSON()
+	initialSize := len(json)
+	metricsCount := len(series)
+
+	payloadBuilder := stream.NewJSONPayloadBuilder(true)
+	var totalTime time.Duration
+
+	for i := 0; i < runs; i++ {
+		start := time.Now()
+		payloadBuilder.Build(series)
+		totalTime += time.Since(start)
+	}
+	avgTime := int64(totalTime) / int64(runs)
+	speed := float64(initialSize) / (float64(avgTime) / float64(time.Second))
+	metricRate := int(float64(metricsCount) / (float64(avgTime) / float64(time.Second)))
+	megabyte := float64(1024 * 1024)
+	fmt.Printf("inputSize: %d bytes \t # of metrics: %d \t tags: %d \t points: %d \t avg duration: %s \t throughput: %f MB/sec \t metrics/sec: %d\n", initialSize, metricsCount, tags, points, fmt.Sprint(time.Duration(avgTime)), speed/megabyte, metricRate)
+}
+
+// Uncomment to run these benchmarks. These are non-standard benchmarks to collect custom metrics
```

Comment:
Would it work if the file was conditional on a build tag instead?

Reply:
Good question. Do we have an existing tag we can leverage? I'm not sure what the team best practice is for this.

Reply:
This might be a good idea, we could move these over to a separate file, and use an `+ optional_benchmarks` tag or something.